apis/config/v1beta1/configuration_types.go

```diff
@@ -349,3 +352,28 @@ type ClusterQueueVisibility struct {
 	// Defaults to 10.
 	MaxCount int32 `json:"maxCount,omitempty"`
 }
+
+// +kubebuilder:validation:Enum=LessThanOrEqualToFinalShare;LessThanInitialShare
+type PreemptionStrategy string
+
+const (
+	LessThanOrEqualToFinalShare PreemptionStrategy = "LessThanOrEqualToFinalShare"
+	LessThanInitialShare        PreemptionStrategy = "LessThanInitialShare"
+)
+
+type FairSharing struct {
+	// enabled indicates whether fair sharing is enabled for all cohorts.
+	// Defaults to false.
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
```suggestion
	Enabled *bool `json:"enabled,omitempty"`
```

Because this is optional.

Reply:
But the entire struct is already a pointer.

Reply:
But this field is not optional https://github.com/kubernetes-sigs/kueue/pull/2070/files#diff-b8a00d3ff4d5cac56b83519c4b98c4f8d27c6f3855c68327842289ed64f87722R418

It's just omitted when false. Maybe I can just remove `omitempty`.

Reply:
> It's just omitted when false. Maybe I can just remove omitempty.

That makes sense!